src/controller/CHIPDeviceController.cpp

```diff
@@ -504,22 +504,19 @@ CHIP_ERROR DeviceController::OpenCommissioningWindowInternal()
 
     if (mCommissioningWindowOption != CommissioningWindowOption::kOriginalSetupCode)
     {
+        // TODO: Salt should be provided as an input or it should be randomly generated when
+        // the PIN is randomly generated.
+        const char * kSpake2pKeyExchangeSalt = "SPAKE2P Key Salt";
         ByteSpan salt(Uint8::from_const_char(kSpake2pKeyExchangeSalt), strlen(kSpake2pKeyExchangeSalt));
         bool randomSetupPIN = (mCommissioningWindowOption == CommissioningWindowOption::kTokenWithRandomPIN);
         PASEVerifier verifier;
 
         ReturnErrorOnFailure(PASESession::GeneratePASEVerifier(verifier, mCommissioningWindowIteration, salt, randomSetupPIN,
                                                                mSetupPayload.setUpPINCode));
 
-        uint8_t serializedVerifier[2 * kSpake2p_WS_Length];
-        VerifyOrReturnError(sizeof(serializedVerifier) == sizeof(verifier), CHIP_ERROR_INTERNAL);
-
-        memcpy(serializedVerifier, verifier.mW0, kSpake2p_WS_Length);
-        memcpy(&serializedVerifier[kSpake2p_WS_Length], verifier.mL, kSpake2p_WS_Length);
-
         AdministratorCommissioning::Commands::OpenCommissioningWindow::Type request;
         request.commissioningTimeout = mCommissioningWindowTimeout;
-        request.PAKEVerifier         = ByteSpan(serializedVerifier);
+        request.PAKEVerifier         = ByteSpan(reinterpret_cast<const uint8_t *>(&verifier), sizeof(verifier));
```

Comment:
This is relying on the exact object layout of `verifier`, which is not guaranteed.... this is why the old code copied into a buffer where we do know the layout.

Reply:
This is fair observation. I assumed that was expected because in the original PASESession::ComputePASEVerifier() code the verifier output of the mPBKDF.pbkdf2_sha256() is used without de-serialization. I'll fix it in both places.

Reply:
This is addressed in a separate PR: #14719